tests/lib/c_lib/src/test_strerror.c

```diff
@@ -32,6 +32,7 @@ ZTEST(test_c_lib, test_strerror)
 	/* do not change errno on success */
 	zassert_equal(4242, errno, "");
 
+#ifndef CONFIG_EXTERNAL_LIBC
```

Comment:
For the heck of it, I thought I would check what happens here on Ubuntu Linux 22.04 and macOS Ventura.

For the invalid value -42:
* `Unknown error -42` (Ubuntu)
* `Unknown error: -42` (macOS)

For the invalid value 4242:
* `Unknown error 4242` (Ubuntu)
* `Unknown error: 4242` (macOS)

I would be happy to update the minimal libc implementation to be a bit more "normal" if it allows this test to be more consistent. Could adjust later, if necessary, in case some other libc / distro variant out there differs.

```
gcc --version
gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
```
gcc --version 
Apple clang version 15.0.0 (clang-1500.0.40.1)
Target: arm64-apple-darwin22.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

Reply:
@keith-packard  #63951 if it helps at all.